plugins/link/dialogs/anchor.js

```diff
@@ -27,15 +27,15 @@ CKEDITOR.dialog.add( 'anchor', function( editor ) {
 			return null;
 		}
 
-		if ( element.is( 'a' ) ) {
+		if ( element && element.data( 'cke-real-element-type' ) === 'anchor' ) {
 			return element;
 		}
 
 		// In case of table cell selection, we want to shrink selection from td to a element.
 		range.shrink( CKEDITOR.SHRINK_ELEMENT );
 		element = range.getEnclosedNode();
 
-		return element.type === CKEDITOR.NODE_ELEMENT && element.is( 'a' ) && element;
+		return element.type === CKEDITOR.NODE_ELEMENT && element && element.data( 'cke-real-element-type' ) === 'anchor';
 	}
```

Comment:
It seems to be working solution. I just extend the if statement to also accept non-fake `a` elements as it was made at the origin solution.